admin/c2cgeoportal_admin/views/layer_groups.py

```diff
@@ -54,7 +66,15 @@ def view(self):
                  request_method='POST',
                  renderer='../templates/edit.jinja2')
     def save(self):
-        return super().save()
+        response = super().save()
+        parent_id = self._appstruct['parent_id'] \
+            if hasattr(self, '_appstruct') and self._appstruct is not None \
+            and self._appstruct.get('parent_id') else None
```

Comment:
I am not a big fan of such indirection... 
We could have state 
has_to_be_registred_in_parent = hasattr(self, '_appstruct') and self._appstruct is not None and self._appstruct.get('parent_id')

Anyway, I am surprised that we have to look for it in appStruct and that it was not objectfied, probably because we are setting an int where model look for a list... Is it ?

Reply:
Note that this is strictly equivalent to:
```
parent_id = getattr(self, '_appstruct', {}).get('parent_id', None)
```
My question would be : Does this correctly handle validation errors.
If there is a validation error, something like ```self._obj and inspect(self._obj).persistent``` should be False.

Is it possible to add a test with a validation error, that the form correctly render with validation error message.

Reply:
It does correctly handle validation error as if there is a validation error, cstruct is empty, that is the magic.

Reply:
OK so adding a comment could be a good idea

Reply:
So here i updated with christophe's suggerstion and add a comment